luarules/gadgets/unit_seismic_dots_and_advanced_radar.lua

```diff
@@ -0,0 +1,296 @@
+
+local gadget = gadget ---@type Gadget
+
+function gadget:GetInfo()
+	return {
+		name      = "Seismic Dots and Advanced Radar",
+		desc      = "When in Seismic LOS, reveal a wobbling radar dot. And allow for custom wobble if in LOS of an advanced radar",
+		author    = "Kyle Anthony Shepherd (Itanthias)",
+		date      = "Aug 7, 2025",
+		license   = "GNU GPL, v2 or later, and anyone who uses this gadget has to email me a rabbit with a sword at kyleanthonyshepherd@gmail.com",
+		layer     = -1, -- to run after most gadgets have handled their stuff
+		enabled   = Spring.GetModOptions().sensor_rework
+	}
+end
+
+if not gadgetHandler:IsSyncedCode() then
+	return false
+end
+
+-- Golden Ratio/Angle magic to create "random" wobble without runs of no wobble
+local goldenAngle = (2-(1 + math.sqrt(5))/2)*math.pi*2
+local goldenSteps = {1,4,7,9,12,14}
+local mRandom = math.random
+local mSqrt = math.sqrt
+
+local spGetUnitAllyTeam = Spring.GetUnitAllyTeam
+local spGetUnitPosition = Spring.GetUnitPosition
+local spTraceRayGroundBetweenPositions = Spring.TraceRayGroundBetweenPositions
+local spIsUnitInRadar = Spring.IsUnitInRadar
+local spSetUnitLosState = Spring.SetUnitLosState
+local spSetUnitLosMask = Spring.SetUnitLosMask
+local spGetGameFrame = Spring.GetGameFrame
+local spSetUnitPosErrorParams = Spring.SetUnitPosErrorParams
+local spGetUnitIsStunned = Spring.GetUnitIsStunned
+
+local allyteamlist = Spring.GetAllyTeamList()
+
+local restoreUnitLOS = {} -- list of units to hand off los control back to engine after X frames
+
+local advancedRadars = {} -- table to hold all advanced radars currently in the game, partitioned by allyTeam
+for key, values in pairs(allyteamlist) do
+	advancedRadars[values] = {} 
+end
+
+local inBasicRadar = {} -- table to hold units currently in basic radar, partitioned by allyTeam
+local basicRadarUpdateRate = 90
```

Comment:
assuming it's 90 frames:
```suggestion
local basicRadarUpdateRate = 3 * Game.gameSpeed
```